### ISSUE TITLE
[Feature Request]:  Add support for ZipMap and Tree Ensemble Classifier

### ISSUE THREAD

**Issue by AbhijithGanesh** [openvinotoolkit/openvino#23778]
### Request Description

I was trying to implement a CatBoost model inferencing via ONNX and ran into this error:

```
RuntimeError: Exception from src/inference/src/cpp/core.cpp:92:
Check 'error_message.empty()' failed at src/frontends/onnx/frontend/src/frontend.cpp:122:
FrontEnd API failed with GeneralFailure:
OpenVINO does not support the following ONNX operations: ai.onnx.ml.ZipMap, ai.onnx.ml.TreeEnsembleClassifier
```

### Feature Use Case

This would extend support for Catboost and Tree Ensemble Classifier

### Issue submission checklist

- [X] The feature request or improvement must be related to OpenVINO

**Comment by AbhijithGanesh** [comment]
![image](https://github.com/openvinotoolkit/openvino/assets/67182544/00e1bc6a-7bdb-461a-91a6-3e7e375fa808)

**Comment by krish1209** [comment]
Hi. You can develop a custom logic for the unsupported operations, or try checking the ONNX opset version. If it is supported in a lower Opset version.

**Comment by AbhijithGanesh** [comment]
I am not sure if it is supported by a lower OpSet version, thats why I wanted to raise this issue and add support

**Comment by AbhijithGanesh** [comment]
@gkrivor Can I work alongside you on this?

**Comment by AyuK03** [comment]
> ### Request Description
> I was trying to implement a CatBoost model inferencing via ONNX and ran into this error:
> 
> ```
> RuntimeError: Exception from src/inference/src/cpp/core.cpp:92:
> Check 'error_message.empty()' failed at src/frontends/onnx/frontend/src/frontend.cpp:122:
> FrontEnd API failed with GeneralFailure:
> OpenVINO does not support the following ONNX operations: ai.onnx.ml.ZipMap, ai.onnx.ml.TreeEnsembleClassifier
> ```
> 
> ### Feature Use Case
> This would extend support for Catboost and Tree Ensemble Classifier
> 
> ### Issue submission checklist
> * [x]  The feature request or improvement must be related to OpenVINO



> ### Request Description
> I was trying to implement a CatBoost model inferencing via ONNX and ran into this error:
> 
> ```
> RuntimeError: Exception from src/inference/src/cpp/core.cpp:92:
> Check 'error_message.empty()' failed at src/frontends/onnx/frontend/src/frontend.cpp:122:
> FrontEnd API failed with GeneralFailure:
> OpenVINO does not support the following ONNX operations: ai.onnx.ml.ZipMap, ai.onnx.ml.TreeEnsembleClassifier
> ```
> 
> ### Feature Use Case
> This would extend support for Catboost and Tree Ensemble Classifier
> 
> ### Issue submission checklist
> * [x]  The feature request or improvement must be related to OpenVINO

I would like to take up this issue.

**Comment by AyuK03** [comment]
.take

**Comment by AyuK03** [comment]
@AbhijithGanesh is this issue still open? I would like to work on it

**Comment by AbhijithGanesh** [comment]
Yeah I am exploring how to add these features. I'm yet to open my draft PR, I'll do that soon

**Comment by mlukasze** [comment]
@gkrivor could you advice, please?
